src/sentry/monitors/models.py

```diff
@@ -284,6 +284,8 @@ class MonitorCheckIn(Model):
     date_added = models.DateTimeField(default=timezone.now)
     date_updated = models.DateTimeField(default=timezone.now)
     attachment_id = BoundedBigIntegerField(null=True)
+    # Holds data about the configuration of the monitor at check-in time such as margin, next_checkin, max runtime
+    config_data = JSONField(null=True)
```

Comment:
Do we have a relatively fixed schema here? If so could we potentially use explicit cols?

Reply:
So I think what we'd like to do here is remember what the monitor configuration is at the time of the checkin, this allows us to compute / store what the *actual* expected checkin time for this checkin was, as well as indicate what the margin of error was.

It would be nice to also store the schedule so we can show that in a tooltip or whatever, or just indicate on the frontend "the schedule at the time of this checkin is different from what what it is now"

Right now we store this in a JSONField on the monitor, so I think it's OK to store it similarly here, I don't expect us to need to query on this, and it gives us some flexibility.

What are you thinking for explicit columns @wedamija? I'm assuming it would be nice to know exactly what data is there, since a JSON column could drift (especially bad considering just how many checkins we have)

Reply:
@davidenwang let me know if this lines up with what you're thinking too

Reply:
Yea I agree, it would be useful to just mirror aspects of the monitor config in this object, and like you said we will probably never need to query on them, just have access later for displaying to the user.

Reply:
It's not just about querying on them, it's about understanding the contents of the field. Json blobs tend to just accumulate fields and be hard to work with. If there's a fixed schema, then I think we'd be better off with real columns so that it's clear how it works. I guess I'd like to understand what the potential fields are, and whether they'll vary at all. Between monitors. If they're always the same, there's not really much reason to use a json blob.

An alternative could be to define a json schema here so that at least it's clear what values we expect.

Reply:
yeaa thats fair, in that case we probably want to add a couple columns to this model. For the start though I think the two we would need is just the `expected_time` and the `margin`. In the future I could see it being useful to add the rest of the monitor config here as potential fields: `schedule`, `schedule_type`, `timezone` etc... 

but for now I can change this to have `expected_time` and `margin` as explicit columns